From f032da4a05d63b57c9c7659185a64d4e2f4240d7 Mon Sep 17 00:00:00 2001
From: George McCollister <george.mccollister@gmail.com>
Date: Thu, 2 Dec 2010 09:31:38 -0600
Subject: [PATCH] check if splice() takes 6 arguments

checking for the presence of splice() isn't sufficient. Older
implimentations only accept 4 arguments while this version of glib
assumes that it takes 6.

Signed-off-by: George McCollister <george.mccollister@gmail.com>
---
 configure.in |   17 ++++++++++++++++-
 1 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index f6d80fb..bb7266c 100644
--- a/configure.in
+++ b/configure.in
@@ -969,7 +969,22 @@ AC_CHECK_FUNCS(chown lchmod lchown fchmod fchown link statvfs statfs utimes getg
 AC_CHECK_FUNCS(getmntent_r setmntent endmntent hasmntopt getmntinfo)
 # Check for high-resolution sleep functions
 AC_CHECK_FUNCS(nanosleep nsleep)
-AC_CHECK_FUNCS(splice)
+
+AC_CACHE_CHECK([for splice()],
+  [ac_cv_func_splice],
+  [AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#define _GNU_SOURCE
+#include <stdlib.h>
+#include <fcntl.h>
+  ]],[[
+    splice(0, NULL, 0, NULL, (size_t)0, (unsigned int)0);
+  ]])],
+  [ac_cv_func_splice=yes],
+  [ac_cv_func_splice=no])])
+
+AS_IF([test "x$ac_cv_func_splice" = "xyes"],[
+  AC_DEFINE([HAVE_SPLICE],[1],[If we have splice])
+])
 
 AC_CHECK_HEADERS(crt_externs.h)
 AC_CHECK_FUNCS(_NSGetEnviron)
-- 
1.7.1

